akka-cluster/src/main/scala/akka/cluster/ClusterDaemon.scala

```diff
@@ -190,8 +190,7 @@ private[cluster] final class ClusterDaemon(settings: ClusterSettings) extends Ac
   override def postStop(): Unit = {
     clusterShutdown.trySuccess(Done)
     if (Cluster(context.system).settings.RunCoordinatedShutdownWhenDown) {
-      // run the last phases e.g. if node was downed (not leaving)
-      coordShutdown.run(Some(CoordinatedShutdown.PhaseClusterShutdown))
+      coordShutdown.run()
```

Comment:
Would it make sense to makes this configurable? I think it could be done in a separate PR though. 

So, instead of having `run-cs-when-down = on` we could have `run-cs-phase-when-down = "service-stop" # leave blank to run all`. This could also apply to `run-as-jvm-shutdown-hook = on`. So that the two triggers CS has built-in can be configured to run from different phases.

Reply:
When would you ever want to run a few of the phases, and why would you then just skip a few of the first phases? For such special cases I think a task itself should look at some external state to decide what to do, e.g. do nothing. The Cluster state is easily accessible so other tasks can check that.

We could add a field to CS telling that it was started by JVM hook, in case a task want to do different things for that case.

Reply:
> We could add a field to CS telling that it was started by JVM hook, in case a task want to do different things for that case.

Knowing the cause for CS.run would be very useful. In Play/Lagom in particular, knowing if it's a SIGTERM in or a node downing in production vs an Application reload in DevMode could be of help.

Reply:
I think it's a good idea to be able to pass a `ShutdownCause` to the `run()` method, and that is then accessible. We can define a marker interface `ShutdownCause` and a few predefined causes, such as ClusterDowning, ClusterLeaving, JvmExit. Applications and libraries can define other causes (i.e. ShutdownCause will not be sealed).

Reply:
In Play/Lagom using CS has to be done with some care. We have to add it in replacement of `system.terminate()` so that it is used, but the default phase is `actor-system-terminate` so that behavior doesn't change. This lack of change of behavior is to ensure that play users that also embed an akka cluster node may already have cluster shutdown code and using `CS.run` would interfere.

The replacement, though, allows for [overriding the default behavior](https://github.com/playframework/playframework/pull/8053/files#diff-8e1dfac265f0cfecaa72a78c9a66f889R152) so that Lagom (or play users with akka cluster) can replace their current, obsolete code with Akka's default cluster shutdown.